src/deploy/osp_deployer/director.py

```diff
@@ -1256,13 +1256,18 @@ def setup_nic_configuration(self):
             cmds.append('sed -i -r \'s/(^\s*' + setting_name +    # noqa: W605
                         ':\s*).*/\\1' + setting_value + '/\' ' + remote_file)
 
+        # if smart NICs are used then use smart configuration for contoller and compute nodes
+        if self.settings.enable_smart_nic == True and self.get_sriov_interfaces() == 2:
+            cmds.append('sed -i s/controller.yaml/controller_smart.yaml/ ' + remote_file)
+            cmds.append('sed -i s/dellcompute_sriov.yaml/dellcompute_sriov_smart.yaml/ ' + remote_file)
+
         # Execute the commands
         for cmd in cmds:
             self.run(cmd)
 
     def setup_sriov_nic_configuration(self):
         logger.debug("setting SR-IOV environment")
-        # Get seetings from .ini file
+        # Get settings from .ini file
         ini_nics_settings = self.settings.get_curated_nics_settings()
 
         cmds = []
```

Comment:
The code below really should call get_sriov_interfaces() instead of duplicating that code here